records/2023-Q1-activity-report.md

```diff
@@ -0,0 +1,418 @@
+  has not been installed first
+  ([coursier#2714](https://github.com/coursier/coursier/pull/2714)).
+
+### sbt
+
+For Scala 2 and Scala 3.
+
+[sbt](https://scala-sbt.org/) is the most used build tool in the Scala community.
+
+We have set up a CI workflow to generate native binaries for macOS ARM64
+([sbtn-dist#4](https://github.com/sbt/sbtn-dist/pull/4),
+[sbt#7149](https://github.com/sbt/sbt/pull/7149)). These binaries will be
+available starting from version 1.9.0 of sbt.
+
+sbt server is used by IntelliJ and Metals to import an sbt build and trigger compilations. We are continuously working on making sbt server more reliable for the user. This quarter we fixed the following bugs:
+- Deadlock in sbt server, fixed in [#7191](https://github.com/sbt/sbt/pull/7191)
+- Forwarding Java diagnostics through BSP, fixed in [#7142](https://github.com/sbt/sbt/pull/7142)
+
+### TASTy-MiMa
+
+For Scala 3.
+
+TASTy-MiMa is a new tool that checks TASTy-compatibility across library versions.
+It is similar to MiMa, which works at the binary compatibility level.
+
+The project can be found at [https://github.com/scalacenter/tasty-mima](https://github.com/scalacenter/tasty-mima).
+It is based on tasty-query (see below).
+
+We finished the first implementation of TASTy-MiMa, and published it along with an sbt plugin.
+It checks all the possible TASTy incompatibilities across API versions.
+
+A few library maintainers have tried the early versions and reported a number of issues.
+We fixed all the issues raised so far, and published a few follow-up versions.
+
+This completes the initial development of TASTy-MiMa.
+We still intend to encourage library maintainers to use it.
+We have also been discussing whether and how to present a unified front-end on top of MiMa and TASTy-MiMa, to simplify their joint usage.
+
+### TASTy Manipulation Library (tasty-query)
+
+For Scala 3.
+
+tasty-query is a library to read semantic information from Scala 3 classpaths.
+It reads Scala .tasty files, Scala 2 pickles and Java .class files, and presents the semantic information they contain in a unified API.
+It is the basis for TASTy-MiMa (see above) and is used as part of the Scala 3 debugger in VS Code.
+Over time, it should become an essential tool for any static analysis involving Scala code.
+
+The project can be found at [https://github.com/scalacenter/tasty-query](https://github.com/scalacenter/tasty-query).
+
+In this quarter, we fixed a number of issues prompted by feedback on the early versions of TASTy-MiMa.
+An EPFL student is also working on a TASTy Checker based on tasty-query, which is used to further improve the robustness of the library.
+
+### Scala Toolkit
+
+For Scala 2 and 3.
+
+- In [toolkit#4](https://github.com/scala/toolkit/pull/4) we included sttp's uPickle integration. This enables the `asJson[Foo]` response handler.
+- [toolkit#8](https://github.com/scala/toolkit/pull/8) (part of migration to `org.scala-lang` organisation)
```

Comment:
Should this go into the already existing `Scala Toolkit` section in `Documentation and Education`?

Reply:
Good point, I did that in deb55a765acbe5a121a4a1b8b7537265006dd20a.